test/README.md

```diff
@@ -1,95 +1,296 @@
+In a convenient directory, create a folder called auth to store your keys and certs (*I have used /home/azureuser/auth*)
+~~~sh
+mkdir auth
+cd auth
+# Create a private key
+openssl genrsa -out server.key  2048
+# Create a csr
+openssl req -new -key server.key -out server.csr
+# Create a cert from the csr
+openssl x509 -req -days 365 -in server.csr -signkey server.key -out server.crt
+
+# At the end of these steps, you have 3 files in your auth directory - server.key (the  
+# key), server.csr (the certificate signing request) and, server.crt (the certificate)
+~~~
+
+###### Setup native basic auth
+~~~ sh
+sudo apt-get install apache2-utils
+# Create password file
+: | sudo tee htpasswd
+# Create username and password. Replace the username and password strings 
+# below with a user/password combo that you like
+echo "password" | sudo htpasswd -iB htpasswd username
+~~~
+
+###### Configure and run the local docker repository
+~~~ sh
+# Set up the docker registry to listen on port 5000 and to restart automatically.
+sudo docker run -d  -p 5000:5000 \
+  --name registry  \
+  -v /home/azureuser/auth/:/etc/security \
+  -e REGISTRY_HTTP_TLS_CERTIFICATE=/etc/security/server.crt  \
+  -e REGISTRY_HTTP_TLS_KEY=/etc/security/server.key \
+  -e REGISTRY_AUTH=htpasswd   \
+  -e REGISTRY_AUTH_HTPASSWD_PATH=/etc/security/htpasswd   \
+  -e REGISTRY_AUTH_HTPASSWD_REALM="Registry Realm"  \
+  --restart always  \
+  registry:2
+
+# Run docker ps to validate that the registry came up successfully
+sudo docker ps
+# If all goes well, you will see the something like below when you run Docker ps.
+# 59867fe47a5b   registry:2   "/entrypoint.sh /etc…"   6 seconds ago   Up 6 seconds   0.0.0.0:5000->5000/tcp, :::5000->5000/tcp   registry
+
+# Run docker login to validate setup. This step will cache your credentials on the host
+# and will be required later when you build and push docker images
+~~~
+
+##### Install .NET
+End-to-end tests are written in .NET Core and run with the `dotnet test` command, so you need to [install .NET Core SDK](https://docs.microsoft.com/en-us/dotnet/core/install/sdk). *Note: You will have to install .NET version 3.1*
+
+#### Cloud side setup for the tests
+##### IoT Hub
+If you don't already have an existing IoTHub, create one. There is no special configuration required, except for making sure that your IoTHub is enabled for public access.
+
+##### Device Provisioning Service (DPS)
+Create a DPS instance. A subset of the end to end tests will use DPS for testing device provisioning scenarios. 
+###### Create enrollment group for symmetric key based enrollment
+In your DPS instance, create new enrollment group. For this enrollment group, set the attestation type to be symmetric key, toggle IoT Edge Device to true, select your iothub by clicking on Link a new Iot Hub, and select the access policy to be iotHubOwner
+		

```

Comment:
I think this would be an opportunity to point to a corresponding doc: 
https://docs.microsoft.com/en-us/azure/iot-edge/how-to-auto-provision-symmetric-keys?view=iotedge-2020-11&tabs=windows

Reply:
Good point. Added.